draft-ietf-quic-tls.md

```diff
@@ -255,142 +205,203 @@ A simplified TLS 1.3 handshake with 0-RTT application data is shown in
+encryption level are the same in QUIC as in TLS over TCP:
+
+- CRYPTO frames MAY appear in packets of any encryption level.
+- CONNECTION_CLOSE and CRYPTO_CLOSE MAY appear in packets of any
+  encryption level other than 0-RTT.
+- PADDING frames MAY appear in packets of any encryption level.
+- ACK frames MAY appear in packets of any encryption level, but
+  MUST only acknowledge packets which appeared in that encryption
+  level.
+- STREAM frames MAY appear in the 0-RTT and 1-RTT levels.
+- All other frame types MUST only appear at the 1-RTT levels.
+
+Because packets may be reordered on the wire, QUIC uses the packet
+type to indicate which level a given packet was encrypted
+under [TODO: Table needed here?]. When multiple packets of
+different encryptions need to be sent, endpoints SHOULD use
+compound packets to send them in the same UDP datagram.
 
 
 ## Handshake and Setup Sequence
 
 The integration of QUIC with a TLS handshake is shown in more detail in
-{{quic-tls-handshake}}.  QUIC `STREAM` frames on stream 0 carry the TLS
-handshake.  QUIC performs loss recovery {{QUIC-RECOVERY}} for this stream and
-ensures that TLS handshake messages are delivered in the correct order.
+{{quic-tls-handshake}}.
 
 ~~~
-    Client                                             Server
+Client                                                   Server
 
-@H QUIC STREAM Frame(s) <0>:
-     ClientHello
-       + QUIC Extension
-                            -------->
-                        0-RTT Key => @0
-
-@0 QUIC STREAM Frame(s) <any stream>:
-   Replayable QUIC Frames
-                            -------->
-
-                                      QUIC STREAM Frame <0>: @H
-                                               ServerHello
-                                  {TLS Handshake Messages}
-                            <--------
-                        1-RTT Key => @1
-
-                                           QUIC Frames <any> @1
-                            <--------
-@H QUIC STREAM Frame(s) <0>:
-     (EndOfEarlyData)
-     {Finished}
-                            -------->
-
-@1 QUIC Frames <any>        <------->      QUIC Frames <any> @1
-~~~
-{: #quic-tls-handshake title="QUIC over TLS Handshake"}
+<CRYPTO_HS[ClientHello]>  --------->
 
-In {{quic-tls-handshake}}, symbols mean:
+(STREAM[0-RTTData])    --------->
 
-* "<" and ">" enclose stream numbers.
+                       <---------   <ACK,
+                                        CRYPTO_HS[ServerHello]>
 
-* "@" indicates the keys that are used for protecting the QUIC packet (H =
-  handshake, using keys from the well-known cleartext packet secret;
-  0 = 0-RTT keys; 1 = 1-RTT keys).
+                       <---------  {CRYPTO_HS[
+                                           EncryptedExtensions,
+                                                Certificate,
+                                          CertificateVerify,
+                                                  Finished]}
```

Comment:
given that we've got 0-rtt data, it would be super helpful to show a 0.5rtt response..

Reply:
Done.